eslint/eslint-patch/src/eslint-bulk-suppressions/bulk-suppressions-patch.ts

```diff
@@ -15,265 +14,277 @@ interface ISuppression {
-          `"rule" property in suppression is not a string: ${JSON.stringify(suppression, null, 2)}`
-        );
-      return true;
-    })
+    let currentFolder: string = normalizedFileFolderPath;
+    currentFolder; // 'something'.substring(0, -1) is ''
+    currentFolder = currentFolder.substring(0, currentFolder.lastIndexOf('/'))
   ) {
-    throw new Error(
-      `Invalid suppression in "suppressions" array: ${JSON.stringify(json.suppressions, null, 2)}`
-    );
+    const cachedEslintrcFolderPath: string | undefined = eslintrcPathByFileOrFolderPath.get(currentFolder);
+    if (cachedEslintrcFolderPath) {
+      return cachedEslintrcFolderPath;
+    }
+
+    pathsToCache.push(currentFolder);
+    for (const eslintrcFilename of ESLINTRC_FILENAMES) {
+      if (fs.existsSync(`${currentFolder}/${eslintrcFilename}`)) {
+        eslintrcFolderPath = currentFolder;
+        break;
+      }
+    }
   }
-  return true;
-}
 
-function readSuppressionsJson(fileAbsolutePath: string): IBulkSuppressionsJson {
-  const eslintrcDirectory: string = findEslintrcDirectory(fileAbsolutePath);
-  const suppressionsPath: string = path.join(eslintrcDirectory, '.eslint-bulk-suppressions.json');
-  let suppressionsJson: IBulkSuppressionsJson = { suppressions: [] };
-  try {
-    const fileContent: string = fs.readFileSync(suppressionsPath, 'utf-8');
-    suppressionsJson = JSON.parse(fileContent);
-
-    if (!validateSuppressionsJson(suppressionsJson)) {
-      console.warn(
-        `Unexpected file content in .eslint-bulk-suppressions.json. JSON expected to be in the following format:
-{
-  suppressions: {
-      file: string;
-      scopeId: string;
-      rule: string;
-  }[];
-}
-Please check file content, or delete file if suppressions are no longer needed.
-`
-      );
-      suppressionsJson = { suppressions: [] };
+  if (eslintrcFolderPath) {
+    for (const checkedFolder of pathsToCache) {
+      eslintrcPathByFileOrFolderPath.set(checkedFolder, eslintrcFolderPath);
     }
-  } catch {
-    // Do nothing and let JSON.parse() log the error. suppressionsJson will stay as the initialized value.
+
+    return eslintrcFolderPath;
+  } else {
+    throw new Error(`Cannot locate an ESLint configuration file for ${normalizedFilePath}`);
   }
-  return suppressionsJson;
 }
 
-function shouldWriteSuppression(fileAbsolutePath: string, suppression: ISuppression): boolean {
-  if (process.env.ESLINT_BULK_SUPPRESS === undefined) return false;
-
-  if (isSuppressed(fileAbsolutePath, suppression)) return false;
+const suppressionsJsonByFolderPath: Map<string, IBulkSuppressionsConfig> = new Map();
+function getSuppressionsConfigForEslintrcFolderPath(eslintrcFolderPath: string): IBulkSuppressionsConfig {
+  let suppressionsConfig: IBulkSuppressionsConfig | undefined =
+    suppressionsJsonByFolderPath.get(eslintrcFolderPath);
+  if (!suppressionsConfig) {
+    const suppressionsPath: string = `${eslintrcFolderPath}/${SUPPRESSIONS_JSON_FILENAME}`;
```

Comment:
You are describing the scenario where VSCode's ESLint instance persists for a couple hours/days? I'm not sure I understand what scenarios the "suppressionsJsonByFolderPath" Map cache will last a long time since it's in-memory.

Reply:
Added functionality to reload the suppressions list every 10 seconds if running inside the VSCode extension.